File: rest-common/src/main/java/org/chronopolis/rest/models/DepositorCreate.java
package org.chronopolis.rest.models;

import org.hibernate.validator.constraints.NotBlank;

import javax.validation.Valid;
import java.util.List;

/**
 * For use in POST /api/depositors or creation through a form
 *
 * @author shake
 */
public class DepositorCreate {

    @NotBlank
    private String namespace;

    @NotBlank
    private String sourceOrganization;

    @NotBlank
    private String organizationAddress;

    @Valid
    private List<DepositorContactCreate> contacts;

    private List<String> replicatingNodes;

    public String getNamespace() {
        return namespace;
    }

    public DepositorCreate setNamespace(String namespace) {
        this.namespace = namespace;
        return this;
    }

    public String getSourceOrganization() {
        return sourceOrganization;
    }

    public DepositorCreate setSourceOrganization(String sourceOrganization) {
        this.sourceOrganization = sourceOrganization;
        return this;
    }

    public String getOrganizationAddress() {
        return organizationAddress;
    }

    public DepositorCreate setOrganizationAddress(String organizationAddress) {
        this.organizationAddress = organizationAddress;
        return this;
    }

    public List<DepositorContactCreate> getContacts() {
        return contacts;
    }

    public DepositorCreate setContacts(List<DepositorContactCreate> contacts) {
        this.contacts = contacts;
        return this;
    }

    public List<String> getReplicatingNodes() {
        return replicatingNodes;
    }

    public DepositorCreate setReplicatingNodes(List<String> replicatingNodes) {
        this.replicatingNodes = replicatingNodes;
        return this;
    }
}

File: rest-models/src/main/kotlin/org/chronopolis/rest/models/create/DepositorCreate.kt
package org.chronopolis.rest.models.create

import org.chronopolis.rest.models.Depositor
import org.chronopolis.rest.models.DepositorContact
import javax.validation.constraints.NotBlank
import javax.validation.Valid

/**
 * Request used to create a [Depositor]
 *
 * @property namespace the namespace to use for the [Depositor]
 * @property sourceOrganization the organizational name of the [Depositor]
 * @property organizationAddress an address for the [Depositor]
 * @property contacts an initial list of [DepositorContact]s
 * @property replicatingNodes an initial list of nodes, identified by their name, which this
 * [Depositor] will distribute data to
 * @author shake
 */
data class DepositorCreate(@get:NotBlank var namespace: String = "",
                           @get:NotBlank var sourceOrganization: String = "",
                           @get:NotBlank var organizationAddress: String = "",
                           @get:Valid var contacts: List<DepositorContactCreate> = mutableListOf(),
                           var replicatingNodes: List<String> = mutableListOf())